core/src/main/java/cucumber/runtime/order/RandomOrderType.java

```diff
@@ -0,0 +1,39 @@
+package cucumber.runtime.order;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import cucumber.runtime.CucumberException;
+import gherkin.events.PickleEvent;
+
+public class RandomOrderType extends OrderType {
+
+	private int count;
+	
+	public RandomOrderType() {}
+
+	public RandomOrderType(Map<String, String> orderTypeData) {
+		this(orderTypeData.containsKey(PROPERTY_COUNT) ? 
+				Integer.parseInt(orderTypeData.get(PROPERTY_COUNT)) : 0);
+	}
+
+	public RandomOrderType(int count) {
+		this.count = count;
+	}
+
+	@Override
+	public List<PickleEvent> orderPickleEvents(List<PickleEvent> pickleEvents) {	
+		Collections.shuffle(pickleEvents);
+		return pickleEvents.subList(0, ((count > pickleEvents.size() || count == 0) ? 

```

Comment:
Nifty but way to hard to follow. 

I think this class is doing too much. It's both ordering and selecting pickles. It should only order pickles. By making `--order` and `--count` independent of each other you can remove this complexity all together.

Reply:
Yeah my bad. Though this logic needs to remain, as need to make sure that the `count` is less or equal to the size of the incoming PickleEvents to avoid exception.

Reply:
No worries!